agent/auto-config/auto_config.go

```diff
@@ -388,39 +318,32 @@ func (ac *AutoConfig) resolveHost(hostPort string) []net.TCPAddr {
 	return addrs
 }
 
-// recordAutoConfigReply takes an AutoConfig RPC reply records it with the agent
+// recordResponse takes an AutoConfig RPC response records it with the agent
 // This will persist the configuration to disk (unless in dev mode running without
 // a data dir) and will reload the configuration.
-func (ac *AutoConfig) recordAutoConfigReply(reply *pbautoconf.AutoConfigResponse) error {
-	// overwrite the auto encrypt DNS SANs with the ones specified in the auto_config stanza
-	if len(ac.config.AutoConfig.DNSSANs) > 0 && reply.Config.AutoEncrypt != nil {
-		reply.Config.AutoEncrypt.DNSSAN = ac.config.AutoConfig.DNSSANs
+func (ac *AutoConfig) recordResponse(resp *pbautoconf.AutoConfigResponse) error {
+	serialized, err := pbMarshaler.MarshalToString(resp)
+	if err != nil {
+		return fmt.Errorf("failed to encode auto-config response as JSON: %w", err)
 	}
 
-	// overwrite the auto encrypt IP SANs with the ones specified in the auto_config stanza
-	if len(ac.config.AutoConfig.IPSANs) > 0 && reply.Config.AutoEncrypt != nil {
-		var ips []string
-		for _, ip := range ac.config.AutoConfig.IPSANs {
-			ips = append(ips, ip.String())
-		}
-		reply.Config.AutoEncrypt.IPSAN = ips
+	if err := ac.updateConfigFromResponse(resp); err != nil {
+		return err
 	}
 
-	conf, err := json.Marshal(translateConfig(reply.Config))
-	if err != nil {
-		return fmt.Errorf("failed to encode auto-config configuration as JSON: %w", err)
+	if err := ac.updateTLSFromResponse(resp); err != nil {
+		return err
```

Comment:
small nit: Might make sense to reuse `ac.update(resp)` here? Otherwise, `update` is only used in one place (`restorePersistedAutoConfig`). Or get rid of `ac.update` entirely?

Reply:
Apparently, I missed this in a refactoring. It now calls update too.